qualtran/bloqs/mcmt/controlled_via_and.py

```diff
@@ -0,0 +1,127 @@
+#  Copyright 2024 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+from collections import Counter
+from functools import cached_property
+from typing import TYPE_CHECKING
+
+import numpy as np
+from attrs import frozen
+
+from qualtran import Bloq, Controlled, CtrlSpec
+from qualtran.bloqs.basic_gates import XGate
+from qualtran.bloqs.mcmt.ctrl_spec_and import CtrlSpecAnd
+
+if TYPE_CHECKING:
+    from qualtran import BloqBuilder, SoquetT
+    from qualtran.resource_counting import BloqCountT, SympySymbolAllocator
+
+
+@frozen
+class ControlledViaAnd(Controlled):
```

Comment:
@anurudhp You still didn't add tests for this class - Please create a follow up PR which adds `qualtran/bloqs/mcmt/controlled_via_and_test.py`, adds bloq examples and ties this bloq to the docs infrastructure. Sorry if I merged it too soon.

Reply:
oops, I was about to add, but no worries! will add in a new PR